test/unit_test/deepdoc/parser/test_pdf_garbled_detection.py

```
@@ -454,3 +454,132 @@ def test_cid_with_zero(self):
+    """
+
+    _LATIN = set("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789 .,-")
+
+    def test_english_is_representable(self):
+        _Parser._OCR_ALPHABET = set(self._LATIN)
+        assert _Parser._ocr_can_represent("Minimum edit distance") is True
+
+    def test_russian_is_not_representable(self):
+        # A Latin-only recogniser cannot spell a Cyrillic page.
+        _Parser._OCR_ALPHABET = set(self._LATIN)
+        assert _Parser._ocr_can_represent("О книге как-то иначе") is False
```

Ruff (0.15.21)

[warning] String contains ambiguous `О` (CYRILLIC CAPITAL LETTER O). Did you mean `O` (LATIN CAPITAL LETTER O)?

(RUF001)